rest/blip_sync.go

```diff
@@ -978,12 +978,23 @@ func (ctx *blipSyncContext) setActiveSubChanges(changesActive bool) {
 
 // setUseDeltas will set useDeltas on the blipSyncContext as long as both sides of the connection have it enabled.
 func (ctx *blipSyncContext) setUseDeltas(clientCanUseDeltas bool) {
+	// If either side don't want deltas, log and exit early
+	if !ctx.sgCanUseDeltas || !clientCanUseDeltas {
+		ctx.Logf(base.LevelInfo, base.KeySync, "Disabling deltas for this replication. SG: %t - Client: %t", ctx.sgCanUseDeltas, clientCanUseDeltas)
+		ctx.useDeltas = false
+		return
+	}
+
+	// Both sides want deltas
 
-	shouldUseDeltas := clientCanUseDeltas && ctx.sgCanUseDeltas
-	if !ctx.useDeltas && shouldUseDeltas {
-		ctx.db.DbStats.StatsDeltaSync().Add(base.StatKeyDeltaPullReplicationCount, 1)
+	if ctx.useDeltas {
+		// deltas are already enabled
+		return
 	}
-	ctx.useDeltas = shouldUseDeltas
+
+	ctx.Logf(base.LevelInfo, base.KeySync, "Enabling deltas for this replication")
```

Comment:
I think this should be switched to debug (i.e. if the above log line isn't present, we assume deltas).